packages/core/src/model/heatmap.ts

```diff
@@ -48,29 +48,18 @@ export class HeatmapModel extends ChartModelCartesian {
 	 * @returns Array consisting of smallest and largest values in  data
 	 */
 	getValueDomain() {
-		const data = this.getDisplayData().map((element) => element.value);
-		const limits = extent(data);
-		const domain = [];
-
-		// Round extent values to the nearest multiple of 10
-		// Axis rounds values to multiples of 2, 5, and 10s.
-		limits.forEach((number, index) => {
-			let value = Number(number);
-
-			if (index === 0 && value >= 0) {
-				value = 0;
-			} else if (value % 10 === 0 || value === 0) {
-				value;
-			} else if (value < 0) {
-				value = Math.floor(value / 10) * 10;
-			} else {
-				value = Math.ceil(value / 10) * 10;
-			}
-
-			domain.push(value);
-		});
+		const limits = extent(this.getDisplayData(), (d: any) => d.value);
+		const domain = scaleLinear()
+			.domain(limits as [number, number])
+			.nice()
+			.domain();
+
+		// Ensuring limits start at 0 to make scale look more `nicer`
+		if (domain[0] > 0) {
```

Comment:
is this correct? what if limits or domain include negative values?

Reply:
Yes, this is the same as what I was doing previously:
```js
if (index === 0 && value >= 0) {
	value = 0;
}
```

If the number is positive, it'll set the minimum in the domain to 0 - Cartesian service does something similar but for axis